stdlib/string.lua

```diff
@@ -52,3 +52,41 @@ function string.split(s, sep, pattern)
     s:gsub(pattern, function(c) fields[#fields+1] = c end)
     return fields
 end
+
+--- Splits a string by another string
+-- @param s the string to split
+-- @param sep the boundary string
+-- @param limit If limit is set and positive, the returned array will contain a maximum of limit elements with the last element containing the rest of string. If the limit parameter is negative, all components except the last -limit are returned. If the limit parameter is zero, then this is treated as 1.
+-- @return table
+function string.explode(s, sep, limit)
+    if not sep or sep == "" then return false end
```

Comment:
if `s` or `sep` are required arguments, the standard practice is to throw an error. For example:

```
require 'stdlib/core'
...
function string.explode(s, sep, limit)
    fail_if_missing(s, "missing string argument")
    fail_if_missing(sep, "missing separator character")
```

If you want to be thorough, you could add tests to verify the failure conditions properly throw an error.